crates/red_knot_python_semantic/src/semantic_index/use_def.rs

```diff
@@ -714,32 +714,32 @@ impl<'db> UseDefMapBuilder<'db> {
             .add_and_constraint(self.scope_start_visibility, constraint);
     }
 
-    #[must_use = "A `*`-import visibility constraint must always be negated after it is added"]
-    pub(super) fn record_star_import_visibility_constraint(
+    pub(super) fn record_and_negate_star_import_visibility_constraint(
```

Comment:
I think it would be useful to future-us to add a doc comment to this method explaining what it is doing and why it is safe to do it, in the limited cases where we use this function.

Reply:
Added some docs! Sadly this PR is no longer net-negative in terms of whether it adds or removes lines of code, but so it goes 🥲

Would appreciate a quick review from you or @sharkdp to double-check everything in the doc-comment is accurate!